dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java

```diff
@@ -57,9 +61,11 @@ public class TaskDTO {
 
     public List<ServerResponseDTO> serverResponses = new ArrayList<>();
 
-    public Set<String> dependants = new HashSet<>();
+    public Set<String> dependants = new TreeSet<>();
 
-    public Set<String> dependencies = new HashSet<>();
+    public Set<String> dependencies = new TreeSet<>();
 
     public ConfigurationDTO configuration = new ConfigurationDTO();
+
+    private Map<Transition, Instant> timestamps = new TreeMap<>();
```

Comment:
Why can't this also be a set of transitiontime or set of transitiontimedto?

Reply:
It definitely can be. By accident, you've answered the question I've had at the start of PR :D. Lucky! Will do that and also make it a List as you've pointed out a very good question.